src/Components/Components/src/ComponentBase.cs

```diff
@@ -171,10 +171,25 @@ void IComponent.Attach(RenderHandle renderHandle)
             _renderHandle = renderHandle;
         }
 
+
         /// <summary>
-        /// Method invoked to apply initial or updated parameters to the component.
+        /// Sets parameters supplied by the component's parent in the render tree.
         /// </summary>
-        /// <param name="parameters">The parameters to apply.</param>
+        /// <param name="parameters">The parameters.</param>
+        /// <returns>A <see cref="Task"/> that completes when the component has finished updating and rendering itself.</returns>
+        /// <remarks>
+        /// <para>
+        /// The <see cref="SetParametersAsync(ParameterView)"/> method should be passed the entire set of parameter values each
+        /// time <see cref="SetParametersAsync(ParameterView)"/> is called. It not required that the caller supply a parameter
+        /// value for all parameters that are logically understood by the component.
+        /// </para>
+        /// <para>
+        /// The default implementation of <see cref="SetParametersAsync(ParameterView)"/> will set a value for every property
+        /// decorated with <see cref="ParameterAttribute"/> or <see cref="CascadingParameterAttribute"/> each time
```

Comment:
I'm a bit surprised by this. Maybe it's something we should consider locking down a bit more before 3.0 ships.

I was expecting that we only assign parameter values where `.Cascading == true` to `[CascadingParameter]` targets, and only where `.Cascading == false` to `[Parameter]` targets. I didn't think it was desired that you can overwrite a `[CascadingParameter]` by explicitly supplying a non-cascading value from the parent. The tooling implies you can't (no cascading parameters appear in intellisense) so if the runtime behaves differently from that it's surprising. The `Cascading` flag on `ParameterValue` is populated but apparently not read back anywhere, which seems very odd.

Do you agree? If so I think we could amend the logic in `ComponentProperties.cs` quite easily to behave this way.

Reply:
> I'm a bit surprised by this. Maybe it's something we should consider locking down a bit more before 3.0 ships.

I'll try this out. I'm not trying to say something contraversial. I agree that this should be locked down.